cpp/src/c_api/graph_mg.cpp

```diff
@@ -126,8 +126,8 @@ struct create_graph_functor : public cugraph::c_api::abstract_functor {
           rmm::device_uvector<edge_t>(edge_ids_->size_, handle_.get_stream());
 
         raft::copy<edge_t>(edgelist_edge_ids.data(),
-                           edge_type_ids_->as_type<edge_t>(),
-                           edge_type_ids_->size_,
+                           edge_ids_->as_type<edge_t>(),
+                           edge_ids_->size_,
```

Comment:
Yeah... I think `edge_type_id_t` and `edge_type_ids_` are a bit confusing names. They can be interpreted as edge types and IDs OR IDs (assigned to each edge type) of edge types and may have contributed to this bug.

Won't `edge_type_t` and `edge_types_` be clearer?

@ChuckHastings Based on search under the cugraph/cpp/src, we're mixing `edge_type_t`, `edge_type_id_t`, and even `edge_type_type_t`, should we better be consistent?

Reply:
Yes, we should be consistent.  I noticed this while working on #2997, but I was scrambling in trying to get something out that people could look at before going on vacation that I didn't address this.

I can try and make things more consistent in #2997.  I think for this PR just correcting the behavior is sufficient.